classes/Product.php

```diff
@@ -7070,7 +7070,7 @@ public function setWsPositionInCategory($position)
             'ORDER BY `position`'
         );
 
-        if ($position > count($result)) {
+        if ($position > count($result) && count($result) > 0) {
```

Comment:
You should avoid repetting the same count twice, on big array it will take some times.

And here if your position is 5 in an empty category, you should still get an error that you would skip here

Reply:
Good point,  I changed it.